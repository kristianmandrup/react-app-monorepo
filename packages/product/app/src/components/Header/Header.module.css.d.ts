export const header_bg: string;
export const headerBg: string;
export const header: string;
