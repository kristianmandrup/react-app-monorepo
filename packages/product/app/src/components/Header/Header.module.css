@import '../../../stylesheets/1_Settings/_settings.colors.css';

.header {
  text-align: center;
  font-size: .8em;
  padding-bottom: 1px;
  background-color: var(--secondaryBackground);
}

/*
  Exports the variable to be exposed in CSS Modules (styles object in React)
*/
:export {
  header_bg: var(--secondaryBackground);
}
