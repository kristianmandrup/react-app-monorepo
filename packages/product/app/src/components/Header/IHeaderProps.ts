export interface IHeaderProps {
  title?: string
}
