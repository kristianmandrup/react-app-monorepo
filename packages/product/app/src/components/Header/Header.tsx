import * as React from 'react'
import * as styles from './Header.module.css'
import { IHeaderProps } from './IHeaderProps'

class Header extends React.Component<IHeaderProps, {}> {
  public render() {
    return (
      <header className={styles.header}>
        <h1>Hello, {this.props.title}</h1>
        <h3>Header background Color from PostCSS => var(--headerBackground): {styles.header_bg}</h3>
      </header>
    )
  }
}

export default Header
