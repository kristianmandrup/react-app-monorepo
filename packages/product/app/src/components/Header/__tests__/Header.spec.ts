import testComponentHelper from '../../../../../test/unit/helpers/ComponentHelper'
import Header from './../Header'

describe('<Header />', () => {
    const initialProps = {
        className: ''
    }
    const renderComponent = testComponentHelper(Header)

    describe('@renders', () => {
        it('in default state', () => {
            expect(renderComponent().getHtml())
                .toMatchSnapshot()
        })
    })
})
