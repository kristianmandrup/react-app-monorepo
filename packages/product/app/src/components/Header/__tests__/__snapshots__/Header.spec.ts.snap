// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> @renders in default state 1`] = `
<header
  className="header"
>
  <h1>
    Hello, 
  </h1>
  <h3>
    Header background Color from PostCSS =&gt; var(--headerBackground): 
    header_bg
  </h3>
</header>
`;
